lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -126,11 +135,12 @@ def script
 
         SH
         script << "\n" unless frameworks_by_config.values.all?(&:empty?)
-        frameworks_by_config.each do |config, frameworks|
-          unless frameworks.empty?
+        frameworks_by_config.each do |config, frameworks_with_dsyms|
+          unless frameworks_with_dsyms.empty?
             script << %(if [[ "$CONFIGURATION" == "#{config}" ]]; then\n)
-            frameworks.each do |framework|
-              script << %(  install_framework "#{framework}"\n)
+            frameworks_with_dsyms.each do |framework_with_dsym|
+              script << %(  install_framework "#{framework_with_dsym[:framework]}"\n)
+              script << %(  install_dsym "#{framework_with_dsym[:dSYM]}"\n) unless framework_with_dsym[:dSYM].nil?
```

Comment:
Yes for non-vendored frameworks that are compiled via sources Xcode will produce and copy the dSYM. It will not be `nil` if the vendored framework pod brings the dSYM with it.